### ISSUE TITLE
Better error handling

### ISSUE THREAD

**Issue by vltr** [vltr/middle#4]
Provide better error handling techniques for developers (almost everywhere)

**Comment by ahopkins** [comment]
Perhaps allowing for capturing **all** validation errors and not just the first?

**Comment by vltr** [comment]
@ahopkins yes, this is a limitation for using `attrs` to generate the classes boilerplate code. Basically, it's impossible to have a fine grained control over either to validate one single class or not, you either have everything being validated (which is good only for production - no need to waste resources if your data is somehow wrong) or nothing is validated, based on a global variable - that can lead to racing condition problems.
